core/domain/exp_jobs_one_off.py

```diff
@@ -492,46 +491,45 @@ def map(item):
-
-        for state in exploration.states.itervalues():
-            # result is the list of all the html present in the state.
-            result = exp_services.find_all_values_for_key(
-                'html', state.to_dict())
-
-            for html_data in result:
-                html_data = html_data.encode('utf-8')
-                soup = BeautifulSoup(html_data, 'html.parser')
-
-                for tag in soup.findAll():
-                    # Checking for tags not allowed in RTE.
-                    if tag.name not in allowed_tag_list:
-                        if 'invalidTags' in err_dict:
-                            err_dict['invalidTags'] += [tag.name]
-                        else:
-                            err_dict['invalidTags'] = [tag.name]
-                    # Checking for parent-child relation that are not
-                    # allowed in RTE.
-                    parent = tag.parent.name
-                    if (tag.name in allowed_tag_list and parent not in
-                            allowed_parent_list[tag.name]):
-                        if tag.name in err_dict:
-                            err_dict[tag.name] += [parent]
-                        else:
-                            err_dict[tag.name] = [parent]
-
-        for key in err_dict.iterkeys():
-            err_dict[key] = list(set(err_dict[key]))
-
-        yield('Errors', err_dict)
+        allowed_parent_list = (
+            feconf.RTE_CONTENT_SPEC['RTE_TYPE_TEXTANGULAR']
+            ['ALLOWED_PARENT_LIST'])
+        allowed_tag_list = (
+            feconf.RTE_CONTENT_SPEC['RTE_TYPE_TEXTANGULAR']
+            ['ALLOWED_TAG_LIST'])
+
+        html_list = exploration.get_all_html_content_strings()
+
+        for html_data in html_list:
+            soup = bs4.BeautifulSoup(html_data, 'html.parser')
+
+            for tag in soup.findAll():
+                # Checking for tags not allowed in RTE.
+                if tag.name not in allowed_tag_list:
+                    if 'invalidTags' in err_dict:
+                        err_dict['invalidTags'] += [tag.name]
+                    else:
+                        err_dict['invalidTags'] = [tag.name]
+                # Checking for parent-child relation that are not
+                # allowed in RTE.
+                parent = tag.parent.name
+                if (tag.name in allowed_tag_list) and (
+                        parent not in allowed_parent_list[tag.name]):
+                    if tag.name in err_dict:
+                        err_dict[tag.name] += [parent]
+                    else:
+                        err_dict[tag.name] = [parent]
+
+        for key in err_dict:
+            yield(key, list(set(err_dict[key])))
 
     @staticmethod
     def reduce(key, values):
-        yield (key, values)
+        final_values = []
+        for value in values:
+            item = ast.literal_eval(value)
+            for sub_item in item:
```

Comment:
It took me quite a while to piece together what's going on here, especially since you are using super-generic variable names like "value", "item" and "sub_item".

It sounds like you're essentially getting the union of a bunch of lists. Why not do something like this instead?

    # Combine all values from multiple lists into a single list for that error type.
    yield(key, set().union(*values))

Reference: https://stackoverflow.com/a/2151553

Reply:
@bansalnitish also FYI Travis is failing.

Reply:
Done!
Added `set().union()`, also the tests would pass now!